src/transaction/lock_manager.c

```diff
@@ -5696,6 +5674,43 @@ lock_initialize (void)
 #endif /* !SERVER_MODE */
 }
 
+#if defined(SERVER_MODE)
+/*
+ * lock_finalize_tran_lock_table - Remove lock table resource
+ *
+ * return: nothing
+ *
+ * NOTE: This function deallocates memory space for transaction lock table.
+ */
+static void
+lock_finalize_tran_lock_table (void)
+{
+  LK_TRAN_LOCK *tran_lock;
+  int i;
+
+  /* remove resources */
+  if (lk_Gl.tran_lock_table != NULL)
+    {
+      for (i = 0; i < lk_Gl.num_trans; i++)
+	{
+	  tran_lock = &lk_Gl.tran_lock_table[i];
+	  pthread_mutex_destroy (&tran_lock->hold_mutex);
+	  pthread_mutex_destroy (&tran_lock->non2pl_mutex);
```

Comment:
It works only if all `lk_Gl.tran_lock_table` elements are initialized. Let's say `lk_Gl.num_trans` is 10 and `lock_finalize_tran_lock_table()` is called from `lock_initialize_tran_lock_table`() while initializing `lk_Gl.num_trans[5]`. Then, `tran_lock->hold_mutex` and `tran_lock->non2pl_mutex` of k_Gl.num_trans[6,7, ..] are not yet initialized but are tried to destroy.

Reply:
@wrlawodms 
I considered it, but I missed it as I turned the structure into a separate function. It's corrected!

Reply:
Different from the current code, I also thought about increasing the value of lk_Gl.num_trans for each loop, but I stopped because I was afraid I would use a lot of macros MAX_NTRANS in the code.
what do you think about it?

Reply:
Eventually changed it. I used variable instead of macro.

Reply:
Looks good to me! Could you tell me why do you use a logical variable, `num_trans`, not just use the macro, `MAX_NTRANS`?

Reply:
@wrlawodms 

The 'MAX_NTRANS' macro cannot be changed because it is used elsewhere. If for some reason you need to change the maximum number of transactions only within this function, you must do so one by one (of course not many in this function). So I used local variables.

Reply:
I mean the value of `int num_trans` is not changed through the function. Isn't it?

Reply:
> Looks good to me! Could you tell me why do you use a logical variable, `num_trans`, not just use the macro, `MAX_NTRANS`?

not logical.. sorry -> local

Reply:
> I mean the value of `int num_trans` is not changed through the function. Isn't it?

It is right! But I thought it would be a little more suitable from a maintenance standpoint when only those values change.

Reply:
Makes sense. Thx.